src/controllers/controller.bar.js

```diff
@@ -285,12 +290,13 @@ module.exports = DatasetController.extend({
 		var scale = me._getValueScale();
 		var isHorizontal = scale.isHorizontal();
 		var datasets = chart.data.datasets;
-		var value = +scale.getRightValue(datasets[datasetIndex].data[index]);
+		var value = scale._parseValue(datasets[datasetIndex].data[index]);
 		var minBarLength = scale.options.minBarLength;
 		var stacked = scale.options.stacked;
 		var stack = meta.stack;
-		var start = 0;
-		var i, imeta, ivalue, base, head, size;
+		var start = value.start === undefined ? 0 : value.max >= 0 && value.min >= 0 ? value.min : value.max;
+		var yValue = value.start === undefined ? value.end : value.max >= 0 && value.min >= 0 ? value.max - value.min : value.min - value.max;
```

Comment:
I would not use `yValue` as variable name, its not correct for horizontalBar. I think `size` would be ok (its calculated to this value later, right?)

Reply:
those variable names are never perfect. @simonbrunel what do you think about var name? I know we discussed this when introduce changes to float-bars but maybe opinion was changed

Reply:
I like `size`. But @kurkle , did you notice there's already another variable `size`? We would have to rename it the existing variable to `sizePx` or something. What do you think of that name for the other variable?

Reply:
I think these are equal things, so same variable can be used (and the later calculation omitted).

Reply:
Yes, I think you're right. Good call. That will be much nicer since it will be one less variable

Reply:
@kurle are your sure about that?

```javascript
base = scale.getPixelForValue(start);	
head = scale.getPixelForValue(start + yValue);
size = head - base;
```

I don't think there are equivalent: the current `size` is the size of the bar in "pixel" unit (mapped from the scale), while `yValue` is the size of the bar in "value" unit.

Reply:
@gwyneblaidd I agree with @kurle, the variable name should not reflect any direction (I would personally call it `length`, `amount`, `range` or `diff` but not `size`).

Reply:
@simonbrunel  ok, its not the same unit of measure. So from those you proposed, I vote for `length`.

Reply:
+1 to `length`